src/storage-client/src/types/sources.proto

```diff
@@ -193,18 +193,26 @@ message ProtoKinesisSourceConnection {
 }
 
 message ProtoPostgresSourceConnection {
+  reserved 5;
+  reserved "table_casts";
+
     message ProtoPostgresTableCast {
         repeated mz_expr.scalar.ProtoMirScalarExpr column_casts = 1;
     }
 
+    message ProtoReferencedColumns {
+        repeated string col_names = 1;
+    }
+
     mz_repr.global_id.ProtoGlobalId connection_id = 6;
     mz_storage_client.types.connections.ProtoPostgresConnection connection = 1;
     string publication = 2;
-    ProtoPostgresSourceDetails details = 4;
-    repeated ProtoPostgresTableCast table_casts = 5;
+    ProtoPostgresSourcePublicationDetails details = 4;
+    map<uint64, ProtoPostgresTableCast> table_casts_map = 7;
```

Comment:
I haven't made up my mind if this will be a good suggestion but I'm putting it here for elaboration. An alternative that wouldn't break backward compatibility and wouldn't add a reserved field would be to only add an extra field of `repeated uint64` that contains the active/selected tables.

Reply:
Implemented this, as well as an approach for introducing mixed-version compatibility. idk if it's the best approach but I believe it should work.